dsperse/src/slice/utils/converter.py

```
@@ -421,39 +421,76 @@ def _unzip_file(zip_file: Path, output_dir: Path):
+                        logger.info(f"Extracted {name} from {source_path.name}")
+                        found_metadata = True
 
-        found_metadata = False
-        with zipfile.ZipFile(archive_path, "r") as zipf:
-            for name in zipf.namelist():
-                if name == "metadata.json" or name.endswith("/metadata.json"):
-                    zipf.extract(name, out)
-                    logger.info(f"Extracted {name} from {archive_path.name}")
-                    found_metadata = True
+            if not found_metadata:
+                raise FileNotFoundError(f"No metadata.json found in {source_path}")
```

Ruff (0.15.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)